brainatlas_api/atlas_gen/atlas_scripts/ratatlas.py

```diff
@@ -0,0 +1,364 @@
+    if not isinstance(major["ANC Name"], str):
+        continue
+
+    idn += 1
+    structures.append(
+        {
+            "acronym": major["ANC Name"].replace(" ", "-"),
+            "id": idn,
+            "name": major["ANC Name"],
+            "structure_id_path": [0, idn],
+            "rgb_triplet": [255, 255, 255],
+        }
+    )
+
+
+for i, submajor in submajors.iterrows():
+    # Get an entry in clean hierarchy with this submajor
+    try:
+        entry = clean_hierarchy.loc[
+            clean_hierarchy.submajornum == submajor["SubANC"]
+        ].iloc[0]
+    except Exception as e:
+        print(e)
+        pass
+
+    # Get path
+    idn += 1
+    path = [0, int(entry.majornum), idn]
+
+    # Append
+    structures.append(
+        {
+            "acronym": submajor["SubANC Name"].replace(" ", "-"),
+            "id": idn,
+            "name": submajor["SubANC Name"],
+            "structure_id_path": path,
+            "rgb_triplet": [255, 255, 255],
+        }
+    )
+
+
+for i, region in clean_hierarchy.iterrows():
+    idn += 1
+    if np.isnan(region.submajornum):
+        path = [0, region.majornum, idn]
+
+    else:
+        path = [0, int(region.majornum), int(region.submajornum), idn]
+
+    structures.append(
+        {
+            "acronym": region.abbreviation,
+            "id": idn,
+            "name": region.name,
+            "structure_id_path": path,
+            "rgb_triplet": [255, 255, 255],
+        }
+    )
+
+# save regions list json:
+with open(uncompr_atlas_path / "structures.json", "w") as f:
+    json.dump(structures, f)
+
+
+# ---------------------------------------------------------------------------- #
+#                                 Create MESEHS                                #
+# ---------------------------------------------------------------------------- #
+print(f"Saving atlas data at {uncompr_atlas_path}")
+meshes_dir_path = uncompr_atlas_path / "meshes"
+meshes_dir_path.mkdir(exist_ok=True)
+
+volume = load_labelled_volume(load_any(str(base_data_fld / "annotated.tif")))
+
+root = extract_volume_surface(volume)
+
+write(root, str(meshes_dir_path / "0.obj"))
+
+# First create a mesh for every minor region
+volume_data = load_any(str(base_data_fld / "annotated.tif"))
+for i, region in tqdm(clean_hierarchy.iterrows()):
+    structure = [
+        s for s in structures if s["acronym"] == region["abbreviation"]
+    ][0]
+    savepath = str(
+        meshes_dir_path
+        / f'{structure["id"]}.obj'.replace("/", "-").replace("\\", "-")
+    )
+    if os.path.isfile(savepath):
+        continue
+
+    vol = np.zeros_like(volume_data)
+
+    if not np.isin(np.float(region.minor), volume_data):
+        # print(f'{region.abbreviation} doesnt seem to appear in annotated dataset')
+        continue
+
+    vol[volume_data == np.float32(region.minor)] = 1
+    if np.max(vol) < 1:
+        raise ValueError
+
+    write(extract_volume_surface(Volume(vol)), savepath)
```

Comment:
We should use the write_reference and write_annotation functions so that all stacks are saved in the same way

Reply:
yep